test/unit/Persona/TokensTest.php

```diff
@@ -1323,4 +1355,56 @@ private function getMockHttpClient(array $responses = [])
 
         return $httpClient;
     }
+
+    /**
+     * Create a mock `Tokens` client, with the cache backend and certificate defined in instance variables
+     * for this class.
+     * @return \Talis\Persona\Client\Tokens
+     */
+    private function getMockTokensClientWithFakeCertificate()
+    {
+        /** @var \Talis\Persona\Client\Tokens|\PHPUnit_Framework_MockObject_MockObject $mockClient */
+        $mockClient = $this->getMock(
+            \Talis\Persona\Client\Tokens::class,
+            ['retrieveJWTCertificate'],
+            [
+                [
+                    'userAgent' => 'unittest',
+                    'persona_host' => 'localhost',
+                    'cacheBackend' => $this->cacheBackend,
+                ]
+            ]
+        );
+
+        $mockClient->expects($this->once())
+            ->method('retrieveJWTCertificate')
+            ->willReturn($this->publicKey);
+
+        return $mockClient;
+    }
+
+    /**
+     * Creates a fake JWT with realistic-looking claim data.
+     *
+     * Note: if you supply the `sub` claim, be sure to add it to the `scopes` claim, and
+     * specify it as the `aud` claim (Persona assigns this to the client ID, incorrectly).
```

Comment:
have we raised a seperate ticket to correct this if it is wrong in persona at the moment?

Reply:
I don’t know - long standing issue. I’ll ensure one is raised.

Reply:
I also thought that perhaps I could drop aud from here, so we don’t have to return and correct the comment.

Reply:
https://github.com/talis/platform/issues/3672 - there's work to do here, but I'll remove the comment to start with.